grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -500,35 +500,60 @@ class StockMovementService {
 
     void removeStockMovementItem(String id) {
         RequisitionItem requisitionItem = RequisitionItem.get(id)
-        removeRequisitionItem(requisitionItem)
+        ShipmentItem shipmentItem = ShipmentItem.get(id)
+        if (requisitionItem) {
+            removeRequisitionItem(requisitionItem)
+        } else {
+            removeShipmentItem(shipmentItem)
+        }
     }
 
     def getStockMovementItems(String id, String stepNumber, String max, String offset) {
         // FIXME should get stock movement instead of requisition
+        Shipment shipment = Shipment.get(id)
```

Comment:
I don't know how I feel about this here. Let's move this into the else (line 542) since it's not used outside of that scope and it won't be tempting to use it outside of the scope.